src/foam/nanos/auth/ServiceProvider.js

```diff
@@ -57,7 +57,10 @@ foam.CLASS({
       name: 'inherentPermissions',
       javaGetter: 'return new String[] { "serviceprovider.read." + getId() };',
       factory: function() {
-        return [ 'serviceprovider.read.' + this.id ];
+        return [
+          'serviceprovider.read.' + this.id,
+          'serviceproviderdao.read.' + this.id
```

Comment:
Why is this required, and this is only for javascript/client, as the inherit implements a javaGetter which supersedes this.

Reply:
User.findSpid(x) is called in various places and yes, I was just about to add it into the javagetter.

Reply:
So it seems likely that when giving someone "serviceprovider.read" permission we should also give "serviceproviderdao.read" to the same user.